scripts/sanitycheck

```diff
@@ -3478,131 +3736,84 @@ def main():
             debug(
                 "{:<25} {:<50} {}SKIPPED{}: {}".format(
                     i.platform.name,
-                    i.test.name,
+                    i.testcase.name,
                     COLOR_YELLOW,
                     COLOR_NORMAL,
                     reason))
 
-
-    def native_and_unit_first(a, b):
-        if a[0].startswith('unit_testing'):
-            return -1
-        if b[0].startswith('unit_testing'):
-            return 1
-        if a[0].startswith('native_posix'):
-            return -1
-        if b[0].startswith('native_posix'):
-            return 1
-        if a[0].split("/",1)[0].endswith("_bsim"):
-            return -1
-        if b[0].split("/",1)[0].endswith("_bsim"):
-            return 1
-
-        return (a > b) - (a < b)
-
-    ts.instances = OrderedDict(sorted(ts.instances.items(),
-                               key=cmp_to_key(native_and_unit_first)))
-
-
-
     if options.report_excluded:
-        all_tests = set(get_unique_tests(ts))
+        all_tests = set(get_unique_tests(suite))
         to_be_run = set()
-        for i,p in ts.instances.items():
-            to_be_run.update(p.test.cases)
+        for i,p in suite.instances.items():
+            to_be_run.update(p.testcase.cases)
 
-        if (all_tests - to_be_run):
+        if all_tests - to_be_run:
             print("Tests that never build or run:")
-            for not_run in (all_tests - to_be_run):
+            for not_run in all_tests - to_be_run:
                 print("- {}".format(not_run))
 
         return
 
-
-    if options.save_tests:
-        ts.save_tests(options.save_tests)
-        return
-
     if options.subset:
-
+        #suite.instances = OrderedDict(sorted(suite.instances.items(),
+        #                    key=cmp_to_key(native_and_unit_first)))
         subset, sets = options.subset.split("/")
-        total = len(ts.instances)
+        total = len(suite.instances)
         per_set = round(total / int(sets))
         start = (int(subset) - 1) * per_set
         if subset == sets:
             end = total
         else:
             end = start + per_set
 
-        sliced_instances = islice(ts.instances.items(), start, end)
-        ts.instances = OrderedDict(sliced_instances)
+        sliced_instances = islice(suite.instances.items(), start, end)
+        suite.instances = OrderedDict(sliced_instances)
+
 
-    info("%d tests selected, %d tests discarded due to filters" %
-         (len(ts.instances), len(discards)))
+    if options.save_tests:
+        suite.csv_report(options.save_tests)
+        return
+
+    info("%d test configurations selected, %d configurations discarded due to filters (excluding lexical rules)." %
```

Comment:
There's a long pause after this message until the next message is printed. Might want to add another status message indicating what sanitycheck is doing.

Reply:
also "lexical rules" is confusing, I don't know what this means. I suggest rephrasing this.

Reply:
removed the lexical rules part, will consider adding this to the docs.